### PR TITLE
[release-4.11] OCPBUGS-10708: delete application should delete all part-of resources

### PR DESCRIPTION
**Pull request by debsmita1**
Manual cherry-pick of https://github.com/openshift/console/pull/12604

/kind bug

### CODE REVIEW

**Review by vikram-raj — CHANGES_REQUESTED** [review]
Tests are failing because of error in `application-utils.spec.ts`

`ERROR in /go/src/github.com/openshift/console/frontend/packages/topology/src/utils/__tests__/application-utils.spec.ts
ERROR in /go/src/github.com/openshift/console/frontend/packages/topology/src/utils/__tests__/application-utils.spec.ts(155,5):
TS2554: Expected 2 arguments, but got 1.`

File: frontend/packages/topology/src/utils/__tests__/application-utils.spec.ts [hunk 1]
```diff
@@ -149,6 +149,24 @@ describe('ApplicationUtils ', () => {
       .catch((err) => fail(err));
   });
 
+  it('Should delete all the specific models related to deployment config if the build config is present', async (done) => {
+    const nodeModel = await getTopologyData(MockResources, 'nodejs-with-bc', 'testproject');
+
+    cleanUpWorkload(nodeModel.resource)
```
Comment (vikram-raj) [inline frontend/packages/topology/src/utils/__tests__/application-utils.spec.ts:155]: Expected 2 arguments, but got 1 in `cleanUpWorkload`.

**Review by vikram-raj — COMMENTED** [review]
Verified it on the cluster bot.

https://user-images.githubusercontent.com/2561818/235189035-abc8bc9e-4c12-4bae-96f4-4fc8c9c2dbc4.mov

/lgtm

**Review by jerolimov — APPROVED** [review]
Tested the latest version and still works fine.

/lgtm

**Pull request merged**